lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -403,52 +472,61 @@ sub get_test_results {
 
     eval { $translator->data } if $translator;    # Provoke lazy loading of translation data
 
-    my $test_info = $self->{db}->test_results( $params->{id} );
+    my $test_info;
     my @zm_results;
-    foreach my $test_res ( @{ $test_info->{results} } ) {
-        my $res;
-        if ( $test_res->{module} eq 'NAMESERVER' ) {
-            $res->{ns} = ( $test_res->{args}->{ns} ) ? ( $test_res->{args}->{ns} ) : ( 'All' );
-        }
-        elsif ($test_res->{module} eq 'SYSTEM'
-            && $test_res->{tag} eq 'POLICY_DISABLED'
-            && $test_res->{args}->{name} eq 'Example' )
-        {
-            next;
-        }
-
-        $res->{module} = $test_res->{module};
-        $res->{message} = $translator->translate_tag( $test_res, $params->{language} ) . "\n";
-        $res->{message} =~ s/,/, /isg;
-        $res->{message} =~ s/;/; /isg;
-        $res->{level} = $test_res->{level};
-
-        if ( $test_res->{module} eq 'SYSTEM' ) {
-            if ( $res->{message} =~ /policy\.json/ ) {
-                my ( $policy ) = ( $res->{message} =~ /\s(\/.*)$/ );
-                if ( $policy ) {
-                    my $policy_description = 'DEFAULT POLICY';
-                    $policy_description = 'SOME OTHER POLICY' if ( $policy =~ /some\/other\/policy\/path/ );
-                    $res->{message} =~ s/$policy/$policy_description/;
-                }
-                else {
-                    $res->{message} = 'UNKNOWN POLICY FORMAT';
-                }
+    eval{
+        $test_info = $self->{db}->test_results( $params->{id} );
+        foreach my $test_res ( @{ $test_info->{results} } ) {
+            my $res;
+            if ( $test_res->{module} eq 'NAMESERVER' ) {
+                $res->{ns} = ( $test_res->{args}->{ns} ) ? ( $test_res->{args}->{ns} ) : ( 'All' );
```

Comment:
What does this line of code do? If the module is NAMESERVER and the argument in the msgid is "{ns}" then the value is used if it exists, else the string "All"? Why and when?

Reply:
This is outside of the scope of this pull request.